shared/js/icons_helper.js

```diff
@@ -140,7 +140,16 @@
       if (!iconURL) {
         iconURL = potentialIcon.src;
       }
-      var sizes = Array.from(potentialIcon.sizes);
+
+      var sizes;
+      if (typeof potentialIcon.sizes === 'string') {
+        // According to the webManifest spec, sizes is a
+        // space-separated string
+        sizes = potentialIcon.sizes.split(' ');
+      } else {
+        // But some websites use Arrays or Sets
```

Comment:
The reason you're seeing this is the WebManifestHelper turns the space separated string into a Set. Can we make this consistent so both the WebManifestHelper and _convertToWebManifestIcons() return the same format, rather than adding this conditional here?